api/envoy/service/discovery/v2/csds.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+
+package envoy.service.discovery.v2;
+
+option java_package = "io.envoyproxy.envoy.service.discovery.v2";
+option java_outer_classname = "CsdsProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+
+import "envoy/admin/v2alpha/config_dump.proto";
+import "envoy/api/v2/core/base.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+// CSDS is Client Status Discovery Service. It can be used to get the status of
+// xDS-compliant client from the management server's point of view.
+service ClientStatusService {
+  rpc StreamClientStatus(stream ClientStatusRequest) returns (stream ClientStatusResponse) {
+  }
+
+  rpc FetchClientStatus(ClientStatusRequest) returns (ClientStatusResponse) {
+    option (google.api.http).post = "/v2/discovery:client_status";
+    option (google.api.http).body = "*";
+  }
+}
+
+// Status of a config.
+enum ConfigStatus {
+  // Status info is not available/unknown.
+  UNKNOWN = 0;
+
+  // Management server has sent the config to client and received ACK.
+  SYNCED = 1;
+
+  // Config is not sent.
+  NOT_SENT = 2;
+
+  // Management server has sent the config to client but hasn’t received
+  // ACK/NACK.
+  STALE = 3;
+
+  // Management server has sent the config to client but received NACK.
+  ERROR = 4;
+}
```

Comment:
One question I still have is that status such as stale is useful to obtain, and you want to know the config responsible for this, but do you also want to know the previous synced config? Would CSDS report both?

Reply:
That's a good question. We can potentially add a last_synced_config field. But management server might not have the last_synced_config available if it doesn't keep track of older configs.

Reply:
Could you share both current and stale configs in the config list? With timestamps it should be possible to figure out which is which.

Reply:
Good idea. In that case, no api change needed. It's all up to the management server what configs to send back.